tests/Aspire.Dashboard.Tests/Integration/Playwright/PlaywrightTestsBase.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Playwright;
+using Xunit;
+
+namespace Aspire.Dashboard.Tests.Integration.Playwright;
+
+public class PlaywrightTestsBase : IClassFixture<DashboardServerFixture>, IClassFixture<PlaywrightFixture>, IAsyncDisposable
+{
+    public DashboardServerFixture DashboardServerFixture { get; }
+    public PlaywrightFixture PlaywrightFixture { get; }
+
+    private IBrowserContext? _context;
+
+    public PlaywrightTestsBase(DashboardServerFixture dashboardServerFixture, PlaywrightFixture playwrightFixture)
+    {
+        DashboardServerFixture = dashboardServerFixture;
+        PlaywrightFixture = playwrightFixture;
+    }
+
+    public async Task RunTestAsync(Func<IPage, Task> test)
+    {
+        var page = await CreateNewPageAsync();
+        try
+        {
+            await test(page);
+        }
+        finally
+        {
+            await page.CloseAsync();
+        }
+    }
+
+    private async Task<IPage> CreateNewPageAsync()
+    {
+        _context ??= await PlaywrightFixture.Browser.NewContextAsync(new BrowserNewContextOptions
```

Comment:
This will be using a single context for all the tests in a class. Is that intentional?

Reply:
nvm, that makes sense at least for a single class at least.